lib/control-connection.js

```diff
@@ -95,10 +95,12 @@ ControlConnection.prototype.init = function (callback) {
           var portSeparatorIndex = name.lastIndexOf(']:');
           return addHost(name.substr(1, portSeparatorIndex - 1), name.substr(portSeparatorIndex + 2), eachNext);
         }
-        if (name.indexOf('.') > 0 && name.indexOf(':') > 0) {
+        if (name.indexOf(':') > 0) {
```

Comment:
I might be wrong, but I think the intent of the code is such that if the notation is an ip address with a port, that we add it directly instead of resolving it as DNS (which is why it was failing for you before).  That was my thought given the comment at line 99 (`// IPv4 with port notation`).

Instead wouldn't it be better to split the contact point if it contains a colon into it's host and port name and then resolve the host (the `resolveAll` further below) with DNS and then add it with the port.